App/Sagas/TextileSagas.js

```diff
@@ -315,3 +305,28 @@ export function * removePayloadFile ({data}) {
   }
   yield put(TextileActions.imageRemovalComplete(id))
 }
+
+export function * retryUploadAfterError ({data}) {
+  const { id } = data
+  const items = yield select(TextileSelectors.itemsById, id)
+  for (const item of items) {
+    if (item.remainingUploadAttempts > 0) {
```

Comment:
When we receive an upload error action, this function runs. If `remainingUploadAttempts > 0`, we upload the file again.